packages/vscode-extension/src/debugging/DebugAdapter.ts

```diff
@@ -93,132 +53,84 @@ export class DebugAdapter extends DebugSession {
-        case "Runtime.executionContextCreated":
-          const context = message.params.context;
-          const threadId = context.id;
-          const threadName = context.name;
-          this.sendEvent(new ThreadEvent("started", threadId));
-          this.threads.push(new Thread(threadId, threadName));
-          break;
-        case "Debugger.scriptParsed":
-          const sourceMapURL = message.params.sourceMapURL;
-
-          if (sourceMapURL?.startsWith("data:")) {
-            const base64Data = sourceMapURL.split(",")[1];
-            const decodedData = Buffer.from(base64Data, "base64").toString("utf-8");
-            const sourceMap = JSON.parse(decodedData);
-            const consumer = await new SourceMapConsumer(sourceMap);
-
-            // We detect when a source map for the entire bundle is loaded by checking if __prelude__ module is present in the sources.
-            const isMainBundle = sourceMap.sources.some((source: string) =>
-              source.includes("__prelude__")
-            );
+    this.breakpointController = new BreakpointsController(
+      this.sourceMapController,
+      this.CDPCommunicator
+    );
+  }
 
-            if (isMainBundle) {
-              this.sendCDPMessage("Runtime.evaluate", {
-                expression: "__RNIDE_onDebuggerReady()",
-              });
-            }
-
-            // Expo env plugin has a bug that causes the bundle to include so-called expo prelude module named __env__
-            // which is not present in the source map. As a result, the line numbers are shifted by the amount of lines
-            // the __env__ module adds. If we detect that main bundle is loaded, but __env__ is not there, we use the provided
-            // expoPreludeLineCount which reflects the number of lines in __env__ module to offset the line numbers in the source map.
-            const bundleContainsExpoPrelude = sourceMap.sources.includes("__env__");
-            let lineOffset = 0;
-            if (isMainBundle && !bundleContainsExpoPrelude && this.expoPreludeLineCount > 0) {
-              Logger.debug(
-                "Expo prelude lines were detected and an offset was set to:",
-                this.expoPreludeLineCount
-              );
-              lineOffset = this.expoPreludeLineCount;
-            }
+  private handleIncomingCDPMethods = async (message: any) => {
+    switch (message.method) {
+      case "Runtime.executionContextCreated":
+        const context = message.params.context;
+        const threadId = context.id;
+        const threadName = context.name;
+        this.sendEvent(new ThreadEvent("started", threadId));
+        this.threads.push(new Thread(threadId, threadName));
+        break;
+      case "Debugger.scriptParsed":
+        const sourceMapURL = message.params.sourceMapURL;
+
+        if (sourceMapURL?.startsWith("data:")) {
+          const { consumer, isMainBundle } = await this.sourceMapController.consumeNewSourceMap(
```

Comment:
consume would suggest that the map is gone after the method call. Can we rename to "addSourceMap" or "registerSourceMap"

Reply:
Also it doesn't feel like sourceMapController should really understand the concept of main bundle. It wouldn't hurt to just keep the line that determines isMainBundle value here instead of moving it into source map controller and making this method return an object that carries that information.